.github/workflows/ci.yml

```diff
@@ -3,34 +3,53 @@ name: CI
 on: [push]
 
 jobs:
-  build:
+  NodeCI:
 
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v1
-    - name: Update Composer
-      run: sudo composer self-update
+
     - name: Cache vendor directory
       uses: actions/cache@preview
+      id: vendorcache
       with:
         path: vendor
-        key: ${{ runner.os }}-composer
+        key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
+        restore-keys: |
+          ${{ runner.os }}-composer-
+
+    - name: Update Composer
+      run: sudo composer self-update
+      if: steps.vendorcache.outputs.cache-hit != 'true'
+
     - name: Install Composer dependencies
       run: composer install --no-interaction --no-progress --no-suggest --no-dev --optimize-autoloader --classmap-authoritative --ignore-platform-reqs
+      if: steps.vendorcache.outputs.cache-hit != 'true'
```

Comment:
This will have side effects - that combination of composer flags means that classes added since the last run will not be available to autoload.